core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -498,8 +503,12 @@ private[kafka] class Processor(val id: Int,
             trace("Closing socket connection actively according to the response code.")
             close(selector, curr.request.context.connectionId)
         }
-      } finally {
-        curr = requestChannel.receiveResponse(id)
+      } catch {
+        case NonFatal(e) =>
```

Comment:
Btw, we should think about whether this is appropriate here and other places where it's being used. In particular, it doesn't catch `OutOfMemoryError`, for example. We should think what's the right behaviour in that case.

Reply:
@ijuma Thank you for the review. Since `Processor` catches all exceptions and propagates only `ControlThrowable`, it makes sense to process all exceptions. Have updated all the catch blocks in `SocketServer`.